enterprise/reporting/reporting.go

```diff
@@ -374,6 +384,19 @@ func (r *HandleT) mainLoop(ctx context.Context, clientName string) {
 	r.getMinReportedAtQueryTime = stats.Default.NewTaggedStat(StatReportingGetMinReportedAtQueryTime, stats.TimerType, tags)
 	r.getReportsQueryTime = stats.Default.NewTaggedStat(StatReportingGetReportsQueryTime, stats.TimerType, tags)
 	r.requestLatency = stats.Default.NewTaggedStat(StatReportingHttpReqLatency, stats.TimerType, tags)
+
+	lastReportedAt := time.Now().UTC().Unix() / 60
+	go func() {
+		// for montering reports pileups
+		for {
+			currentMs := time.Now().UTC().Unix() / 60
+			stats.Default.NewTaggedStat(
+				"reporting_metrics_pileup", stats.GaugeType, stats.Tags{"client": clientName},
+			).Gauge(int(currentMs - lastReportedAt))
```

Comment:
I understand that we are capturing lag rather than pileup, which in my head is the amount of messages rather the time difference.

I would also suffix with the unit we are capturing: 
```suggestion
			stats.Default.NewTaggedStat(
				"reporting_metrics_ lag_minutes", stats.GaugeType, stats.Tags{"client": clientName},
			).Gauge(int(currentMs - lastReportedAt))
```

However, is better/more conventional to use seconds for measuring duration in Prometheus
https://prometheus.io/docs/practices/naming/

Reply:
reporting metrics are stored at minute level, I don't think using seconds will make difference.